### DISCUSSION TITLE
For the problem of atomic positions being too close in the calculated configuration

### DISCUSSION THREAD

**Discussion by woyolo990422** [SSCHAcode/python-sscha#127] [Q&A] [answered]
I am calculating structures with high pressure, around 300TPa (I have a pseudopotential that can describe the pressure at 300TPa), with very close atomic positions and bond lengths of only about 0.6 angstroms. Using sscha to save the generated configurations, I found that many configurations have atomic distances of only about 0.1 angstroms, which makes it difficult for me to accurately calculate the interatomic forces. How to control "ensemble.save", The distance that atoms move in their configuration, enabling them to calculate.

**Comment by mesonepigreco** [comment, marked as answer]
Hi, this probably means that the starting "harmonic" dynamical matrix is really bad, so the configurations extracted are not good.

To fix the issue, try to plot the phonon dispersion and understand why this is happening.

As an easy fix, you could try to stiff the dynamical matrix, however, then the SSCHA will require more iterations to converge.

To stiffen the dynamical matrix, simply multiply the force constant by a factor:

```
dyn = CC.Phonons.Phonons("hamonic_dyn", nqirr=X)  # where X is the number of q points
STIFFEN_FACTOR = 2

for i in range(len(dyn.q_tot)):
    dyn.dynmats[i] *= STIFFEN_FACTOR
    
dyn.save_qe("my_new_dyn")
```

In this case, I strongly advise you to start the SSCHA with a few configurations (10-50) and do many steps before increasing the ensemble size and do more accurate calculations when the dynamical matrix becomes more reasonable.

**Reply by Kurufinve** [reply]
Dear Lorenzo, 

Can you say which parameters of SSCHA relaxation (or minimization) is better to change in the case when auxiliary dynamical matrix becomes bad during the SSCHA relaxation (resulting in generation of configurations with very close interatomic positions)?

> In this case, I strongly advise you to start the SSCHA with a few configurations (10-50) and do many steps before increasing the ensemble size and do more accurate calculations when the dynamical matrix becomes more reasonable.

Does it mean that with small number of configurations it is easier to get reasonable auxiliary dynamical matrix?

Sincerely,
Daniil Poletaev

**Reply by mesonepigreco** [reply]
Not necessary, the use of a small number of configuration is to avoid wasting a lot of time for a dynamical matrix that is completely off, and whose gradients are so big that can be obtained with good precision even with few configurations (like when you have atoms that are too close).  In principle it is always better to use more configurations.
